crucible-install.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+
+# Installer Settings
+SYSCONFIG="/etc/sysconfig/crucible"
+DEPENDENCIES="podman git"
+INSTALL_PATH="/opt/crucible"
+GIT_REPO="https://github.com/perftool-incubator/crucible.git"
+GIT_INSTALL_LOG="/tmp/crucible-git-install.log"
+CRUCIBLE_AUTH_FILE="/root/.docker/crucible-client-server.json"
+
+# User Exit Codes
+EC_DEFAULT_EC=1
+EC_FAIL_USER=3
+EC_FAIL_CLONE=4
+EC_FAIL_INSTALL=5
+EC_FAIL_AUTH=6
+EC_FAIL_DEPENDENCY=7
+EC_FAIL_REGISTRY_UNSET=8
+
+function exit_error {
+    # Send message to stderr
+    printf '\n%s\n\n' "$1" >&2
+    # Return a code specified by $2 or 1 by default
+    exit "${2-1}"
+}
+
+function usage {
+    cat <<_USAGE_
+
+    Crucible installer script.
+
+    Usage: $0 --registry <reg> [ opt ]
+
+    --registry <registry/crucible>
+
+    optional:
+        --name <your full name>
+        --email <your email address>
+        --auth-file <authentication file> (default: $CRUCIBLE_AUTH_FILE)
+
+    --help [displays this usage output]
+_USAGE_
+}
+
+function identity {
+    identity="$HOME/.crucible/identity"
+    if [ -e $identity ]; then
+        echo "Sourcing $identity"
+        . $identity
+    else
+        mkdir -p "$HOME/.crucible"
+    fi
+
+    if [ -z "$CRUCIBLE_NAME" ]; then
+        echo "Please enter your full name:"
+        read CRUCIBLE_NAME
+        echo "CRUCIBLE_NAME=\"$CRUCIBLE_NAME\"" >>"$identity"
+    fi
+    if [ -z "$CRUCIBLE_EMAIL" ]; then
+        echo "Please enter your email address:"
+        read CRUCIBLE_EMAIL
+        echo "CRUCIBLE_EMAIL=\"$CRUCIBLE_EMAIL\"" >>"$identity"
+    fi
+}
+
+
+function has_dependency {
+    has_dep=0
```

Comment:
Indentation seems messed up in this function...

Reply:
oops